vars/infra.groovy

```diff
@@ -34,6 +34,11 @@ Object checkout(String repo = null) {
 }
 
 Object checkoutSCM(String repo = null) {
+    // Enable long paths to avoid problems with tests on Windows agents 
+    if (!isUnix()) {
+        bat 'git config core.longpaths true'
```

Comment:
```suggestion
        bat 'git config --global core.longpaths true'
```

because we want this to apply to all git operations not just the checkout of this source (e.g. many tests amy use the git plugins tests that create new repos for tests)

Reply:
From what I've understood, setting it globally requires system permissions, I'm not sure the pipeline have them here.

Reply:
In the long term, I want to set this setting directly in the agents definitions.

Reply:
setting `system` requires admin permissions, `global` is per user.
Whilst you can lock down user access from their own homes (I have done this in the past for Jeknins installs) ci.jenkins.io does not do this as demonstrated by [this change](https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/502/files#diff-9db55821d407770df4fc37f8dcbb12632e46316fbb4dcba2863e54b87533860dR165) which passes

Reply:
Perfect, thanks for the explanation and example!